src/near_web3_extensions.js

```diff
@@ -0,0 +1,58 @@
+const utils = require('./utils')
+
+module.exports = function (web3) {
+	/**
+	 * Formats the input of retrieveNear and converts all values to HEX
+	 * Tweaked from web3 inputTransactionFormatter to accommodate for retrieveNear:
+	 * https://github.com/ethereum/web3.js/blob/42c2414ace4b0b36be8ed4141b6a3fb7a26198ab/packages/web3-core-helpers/src/formatters.js#L195
+	 *
+	 * @method inputRetrieveNearFormatter
+	 * @param {Object} options
+	 * @returns object
+	 */
+	var inputRetrieveNearFormatter = function (options) {
+		// TODO: check that valid near accountId
+
+		// allow both
+		if (options.gas || options.gasLimit) {
+			options.gas = options.gas || options.gasLimit;
+		}
+
+	    options = _convertOptionsToHex(options);
+
+	    // check from, only if not number, or object
+	    if (typeof(options.from) !== 'number' && typeof(options.from) !== 'object') {
+	        options.from = options.from || (this ? this.defaultAccount : null);
+
+	        if (!options.from && typeof(options.from) !== 'number') {
+	            throw new Error('The send transactions "from" field must be defined!');
+	        }
+
+	        options.from = web3.extend.formatters.inputAddressFormatter(options.from);
+	    }
+
+	    return options;
+	};
+
+	var _convertOptionsToHex = function (options) {
+	    ['gasPrice', 'gas', 'value', 'nonce'].filter(function (key) {
+	        return options[key] !== undefined;
+	    }).forEach(function (key) {
+	        options[key] = utils.decToHex(options[key]);
+	    });
+
+	    return options;
+	};
+
+	const extensions = {
+	  property: 'eth',
```

Comment:
Does it make sense to create a new property to put this under? `web3.near.retrieveNear`

Reply:
Or is that too disconnected that everything is under `web3.eth` except for this one method?

Reply:
I don't have a strong opinion. changed it up, though you can see how you like it lol

Reply:
Let's keep it `eth` for now, but it might make sense to switch everything over to `web3.near` instead of `web3.eth`, since we aren't asking for Ethereum blocks or transactions.

Reply:
i'll make an issue for this.